src/components/ProductCard.js

```diff
@@ -0,0 +1,34 @@
+import { useCart } from '../context/cart-context';
+import { Link } from "react-router-dom";
+
+export default function ProductCard(props) {
+
+    const { state: { cartItems }, dispatch } = useCart();
+    const { item: { id, url, name, author, category, price, inStock } } = props;
+
+    return (
+        <div className="card" key={id}>
+            <img src={url} alt={name} />
+            <div className="card-body">
+                <h3 className="card-title">{name}</h3>
+                <p className="card-text">by {author}</p>
+                <p className="card-text">---{category}</p>
+                <p className="card-text">Rs.{price}</p>
+                {
+                    cartItems.findIndex(item => item.id === id) === -1
+                        ? <button
+                            disabled={inStock ? false : true}
+                            style={inStock ? null : { cursor: "not-allowed" }}
```

Comment:
Cursor style could be added as a class in your CSS.

Reply:
done